tests/std/tests/P0896R4_ranges_alg_fill_n/test.cpp

```diff
@@ -9,48 +9,42 @@
 
 #include <range_algorithm_support.hpp>
 
-constexpr void smoke_test() {
-    using ranges::fill_n, ranges::iterator_t;
-    using std::same_as;
+struct instantiator {
+    template <class Out>
+    static constexpr void call() {
+        using ranges::fill_n, ranges::iterator_t;
+        using std::same_as;
 
-    int expected_output[] = {13, 42, 1367};
-    const int value       = 7;
-    {
-        int output[] = {13, 42, 1367};
-        auto result  = fill_n(ranges::begin(output), ranges::distance(output), value);
-        for (const auto& elem : output) {
-            assert(elem == value);
+        const int expected_output[] = {13, 42, 1367};
+        const int value             = 7;
+        {
+            int output[] = {13, 42, 1367};
+            auto result  = fill_n(ranges::begin(output), ranges::distance(output), value);
+            for (const auto& elem : output) {
+                assert(elem == value);
+            }
+            assert(result == ranges::end(output));
         }
-        assert(result == ranges::end(output));
-    }
-    {
-        int output[] = {13, 42, 1367};
-        auto result  = fill_n(ranges::begin(output), 0, value);
-        for (int i = 0; i < 3; ++i) {
-            assert(output[i] == expected_output[i]);
+        {
+            int output[] = {13, 42, 1367};
+            auto result  = fill_n(ranges::begin(output), 0, value);
+            for (int i = 0; i < 3; ++i) {
+                assert(output[i] == expected_output[i]);
+            }

```

Comment:
This could be abbreviated to `assert(ranges::equal(output, expected_output));` (Ditto lines 39-41.)